plugin/documents.py

```diff
@@ -511,6 +512,21 @@ def on_query_completions(self, prefix: str, locations: List[int]) -> Optional[su
             lambda: self._on_query_completions_async(completion_list, locations[0], triggered_manually))
         return completion_list
 
+    def _format_on_paste(self):
+        self.purge_changes_async()
+        clipboard_text = sublime.get_clipboard()
+        sel = self.view.sel()
+        number_of_cursors = len(sel)
+        split_clipboard_text = clipboard_text.split('\n')
+        for index, region in enumerate(sel):
+            look_text = clipboard_text
+            if len(split_clipboard_text) == number_of_cursors:
+                look_text = split_clipboard_text[index]
+            found_region = self.view.find(look_text, region.end(), sublime.REVERSE)
```

Comment:
In

```ts
type MyCustomType = {
    foo: string;
    bar?: string;
}
```

If I copy and paste in the same place `bar?` then it will crash (and split selection into two).

I wonder if this approach can be made robust... It seems quite fragile.

Reply:
And if the whole content is copied and pasted then it won't find the region and do no formatting.

Reply:
[25b7806](https://github.com/sublimelsp/LSP/pull/2311/commits/25b78067a53dd54b09f65a6fcd0e67498d521263) should fix it